src/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInWidgets.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ContentRecord,
+  CubeName,
+  DataModel,
+  MdxSelect,
+  WidgetWithQueryState,
+} from "@activeviam/activeui-sdk";
+import { MdxString } from "@activeviam/mdx";
+import { migrateCalculatedMeasuresInMdx } from "./migrateCalculatedMeasuresInMdx";
+import { produce } from "immer";
+import _mapValues from "lodash/mapValues";
+
+export const migrateCalculatedMeasuresInWidgets = (
+  widgets: ContentRecord,
+  dataModel: DataModel,
+  namesOfCalculatedMeasurestoMigrate: string[],
+): {
+  cubeNames: { [measureName: string]: CubeName };
+  migratedWidgetsRecord: ContentRecord;
+} => {
+  // Create an empty object where each calculated measure used in a saved widget will be added as a key with its cubeName as a value.
```

Comment:
Just to avoid the confusion of "saved widgets" possibly refering to only standalone widgets, as opposed to widgets in saved dashboards:
```suggestion
  // Create an empty object where each calculated measure used in a widget will be added as a key with its cubeName as a value.
```